service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersController.java

```diff
@@ -71,28 +72,46 @@
     private final HearingBookingService hearingBookingService;
     private final Time time;
     private final RequestData requestData;
+    private final ValidateGroupService validateGroupService;
 
     private static final String JUDGE_AND_LEGAL_ADVISOR_KEY = "judgeAndLegalAdvisor";
 
     @PostMapping("/about-to-start")
     public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
+
+        caseDetails.getData().put("dateOfIssue", time.now().toLocalDate());
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .build();
+    }
+
+    @PostMapping("/date-of-issue/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEventValidateDateOfIssue(
```

Comment:
Also this mid event deals with more than just validating the date of issue, maybe there would be a better name for it

Reply:
I'll remove the word validate from the method, it fires after date of issue so makes sense to me